go/teams/chain.go

```diff
@@ -131,16 +131,22 @@ func (t TeamSigChainState) GetUserLogPoint(user keybase1.UserVersion) *keybase1.
 	return &tmp
 }
 
-// GetLastUserLogPointWithPredicate gets the last user logpoint in the series for which the given
-// predicate is true.
-func (t TeamSigChainState) GetLastUserLogPointWithPredicate(user keybase1.UserVersion, f func(keybase1.UserLogPoint) bool) *keybase1.UserLogPoint {
+// GetLastChangeLogPointWithPredicates gets the first user logpoint in the series at which
+// the second predicate is true AFTER the last user logpoint at which the first predicate is true.
+func (t TeamSigChainState) GetLastChangeLogPointWithPredicates(user keybase1.UserVersion,
+	f1 func(keybase1.UserLogPoint) bool, f2 func(keybase1.UserLogPoint) bool) *keybase1.UserLogPoint {
+
 	points := t.inner.UserLog[user]
 	if len(points) == 0 {
 		return nil
 	}
+	var closestCandidate int
 	for i := len(points) - 1; i >= 0; i-- {
-		if f(points[i]) {
-			tmp := points[i].DeepCopy()
+		if f2(points[i]) {
+			closestCandidate = i
+		}
+		if f1(points[i]) {
```

Comment:
This function is a little weird, what happens if I hit `f1` without `f2` ever firing?

Reply:
huh. yeah i guess it should give the same `nil` response. i'll update that.

Reply:
just kidding. i think it should keep going until it finds the switch it's looking for.